misk-hibernate/src/main/kotlin/misk/hibernate/HibernateExceptionMappers.kt

```diff
@@ -0,0 +1,64 @@
+package misk.hibernate
+
+import com.google.common.base.Throwables
+import misk.exceptions.ConflictException
+import misk.web.Response
+import misk.web.ResponseBody
+import misk.web.exceptions.ExceptionMapper
+import misk.web.mediatype.MediaTypes
+import misk.web.toResponseBody
+import okhttp3.Headers
+import org.hibernate.StaleObjectStateException
+import org.hibernate.exception.ConstraintViolationException
+import org.slf4j.event.Level
+import java.sql.SQLIntegrityConstraintViolationException
+import javax.inject.Inject
+import javax.persistence.OptimisticLockException
+
+internal class RetryTransactionExceptionMapper @Inject internal constructor(
```

Comment:
this does not seem like a general mapper for `RetryTransactionException`, but very specific to the actual cause of the root exception. probably needs a better name to make that clear

Reply:
agreed it's not general. when we have another case I'll rewrite this to get the root cause and check if there's a mapper for it. Or maybe `Throwables.getCasualChains()` and keep going down until there's an exception mapper (or up?)